### PR TITLE
[gracenet]:fix dup closed fd cause use of closed network connection e…

### PR DESCRIPTION
**Pull request by danfu3000**
compare error is "use of closed network connection", skip append to file array to childprocess

### CODE REVIEW

**Comment by danfu3000** [comment]
Sorry, the array len maybe need shorter, wait moment...

File: gracenet/net.go [hunk 1]
```diff
@@ -210,13 +210,19 @@ func (n *Net) StartProcess() (int, error) {
...
+		files[fcnt], err = l.(filer).File()
 		if err != nil {
+			if isUseOfClosedError(err) {
+				continue
+			}
 			return 0, err
 		}
-		defer files[i].Close()
+		defer files[fcnt].Close()
+		fcnt++
```
Comment (zhao-kun) [inline gracenet/net.go:226]: Some fds were closed would not be copied to files. so the `files` array will contain some nil objects, I have no idea about the Go standard library how to process the situation when `os.StartProcess` accepted `ProcAttr` argument which has nil objects

File: gracenet/net.go [hunk 2]
```diff
@@ -250,3 +256,14 @@ func (n *Net) StartProcess() (int, error) {
...
+
+// NOTE: It's work on Go 1.X, Go 2.X maybe change
+func isUseOfClosedError(err error) bool {
+	if err == nil {
+		return false
+	}
+	if opErr, ok := err.(*net.OpError); ok {
+		err = opErr.Err
+	}
+	return err.Error() == "use of closed network connection"
```
Comment (xxx7xxxx) [inline gracenet/net.go:269]: Hmm.. This seems not available to me. Is it a definite type to assert.

**Review by xxx7xxxx — APPROVED** [review]

**Review by zhao-kun — APPROVED** [review]
LGTM

**Pull request merged**